exec/query_executor.go

```diff
@@ -79,18 +79,15 @@ func (q QueryExecutor) ScanStructs(i interface{}) error {
 //
 // i: A pointer to a slice of structs.
 func (q QueryExecutor) ScanStructsContext(ctx context.Context, i interface{}) error {
-	val := reflect.ValueOf(i)
-	if !util.IsPointer(val.Kind()) {
-		return errUnsupportedScanStructsType
+	if _, err := checkScanStructsTarget(i); err != nil {
```

Comment:
Is there a reason to do this check both here and in the `Scanner#ScanStructs`?  Same for `ScanVals`?

Reply:
I should clarify this, I think we should just do the check in the single place for each to keep it DRY.

Reply:
I suggest to call `checkScanStructsTarget` (`checkScanValsTarget`) only in corresponding `Scanner` methods
because errors that cause checks to fail seem likely to not occur at runtime (they should be detected early by simple tests in fact, along with errors like invalid SQL) , so it does not matter that check will be done before actual query execution.
Will update PR appropriately